autopilot/contractor.go

```diff
@@ -1173,6 +1184,7 @@ func (c *contractor) renewContract(ctx context.Context, w Worker, ci contractInf
 	// sanity check the endheight is not the same on renewals
 	endHeight := endHeight(cfg, state.period)
 	if endHeight <= rev.EndHeight() {
+		c.logger.Debugw("invalid renewal endheight", "oldEndheight", rev.EndHeight(), "newEndHeight", endHeight, "period", state.period, "bh", cs.BlockHeight)
```

Comment:
This is what happened in the first round of renewals on `arequipa`, this missing log line is why it wasn't obvious so decided to add it. This endheight issue can be the case if blocks are mined in quick succession between updating the period and checking whether contracts need to be renewed. It might enter the renew window within the old period and then the renewal height is the same as the contract's end height.